infrastructure/smart-contracts/contracts/staking-v2/StakingPool.sol

```diff
@@ -110,13 +117,18 @@ contract StakingPool is AccessControlUpgradeable {
     function getStakedKeysCountForUser(
         address user
     ) external view returns (uint256) {
-        return stakedKeysOfOwner[user].length;
+        if(!hasMigratedToKeyAmount[user]){
+            return stakedKeysOfOwner[user].length;
+        }
+        return stakedKeyAmounts[user];
     }
 
     function getStakedAmounts(address user) external view returns (uint256) {
         return stakedAmounts[user];
     }
 
+    // DEPRECATED We no longer track keys by their IDs
+    // TODO update core function
```

Comment:
What needs to change in core for this?

Reply:
We will probably just remove the unused operator V1 functions, but we need to make sure we don't miss anything still in use somewhere.